src/Components/UpcomingTable/UpcomingTable.tsx

```diff
@@ -27,31 +27,67 @@ import {
   ToolbarItem,
   ToolbarToggleGroup,
 } from '@patternfly/react-core';
-import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import {
+  Table,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+  ExpandableRowContent,
+} from '@patternfly/react-table';
+import { TableRow } from "./UpcomingRow";
 import SearchIcon from '@patternfly/react-icons/dist/esm/icons/search-icon';
 import FilterIcon from '@patternfly/react-icons/dist/esm/icons/filter-icon';
 import './upcoming-table.scss';
 
 import { Record } from '../Upcoming/mock_data';
+import { Simulate } from 'react-dom/test-utils';
+import toggle = Simulate.toggle;
 
 interface UpcomingTableProps {
   data: Record[];
   columnNames: {
     name: string;
+    type: string;
     release: string;
     date: string;
   };
+  details?: {
+    summary: string;
+    potentiallyAffectedSystems: number;
+    trainingTicket: string;
+    dateAdded: string;
+    lastModified: string;
+    detailFormat: 0 | 1 | 2 | 3;
+  };
 }
 
 export const UpcomingTable: React.FunctionComponent<UpcomingTableProps> = (
   props
 ) => {
   // Set up repo filtering
   const { data, columnNames } = props;
-  const [searchValue, setSearchValue] = React.useState('');
+  const [searchValue, setSearchValue] = useState('');
   const [dateSelections, setDateSelections] = React.useState<string[]>([]);
   const [releaseSelection, setReleaseSelection] = React.useState('');
 
+  // Set up Expandable table
+  const [expandedRepoNames, setExpandedRepoNames] = useState([]);
+  const setRepoExpanded = (repo: Record, isExpanding = true) =>
+    setExpandedRepoNames((prevExpanded) => {
+      const otherExpandedRepoNames = prevExpanded.filter(
+        (r) => r !== repo.name
+      );
+      return isExpanding
+        ? [...otherExpandedRepoNames, repo.name]
+        : otherExpandedRepoNames;
+    });
+  const isRepoExpanded = (repo: Record) =>
+    expandedRepoNames.includes(repo.name);
```

Comment:
If I'm not mistaken, we don't need this anymore. The `UpcomingRow` handles this now.